### PR TITLE
chore: Add pkg.engines mandates + require node@14

### PR DESCRIPTION
**Pull request by maranomynet**
## What / Why

Making the system requirements [specified in the README](https://github.com/island-is/island.is#prerequisites) explicit

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] ~~I have commented my code, particularly in hard-to-understand areas~~
- [ ] ~~I have made corresponding changes to the documentation~~
- [x] My changes generate no new warnings
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,6 +2,10 @@
   "name": "island.is",
   "version": "0.0.1",
   "license": "MIT",
+  "engines": {
+    "node": ">=12",
+    "yarn": ">=1.20"
```
Comment (maranomynet) [inline package.json:7]: Should we maybe pin down a major version here?

```suggestion
    "node": "^12.20",
    "yarn": "^1.20"
```

Comment (maranomynet) [reply package.json:7]: @eirikurn has been considering pinning the repo to node@^14 at some point in the near future — so this sort of thing should just wait until then.

**Comment by maranomynet** [comment]
These files also need to be updated to use node@14:

* [scripts/ci/Dockerfile](/island-is/island.is/blob/main/scripts/ci/Dockerfile)
* [scripts/ci/Dockerfile.test](/island-is/island.is/blob/main/scripts/ci/Dockerfile.test)

**Comment by eirikurn** [comment]
Let's change both Dockerfiles to use `node:14.16.0-alpine3.11`

**Comment by maranomynet** [comment]
Done!

**Review by eirikurn — APPROVED** [review]
LGTM

**Pull request merged**